src/target/linux.rs

```diff
@@ -268,4 +317,52 @@ mod tests {
             dealloc(ptr, layout);
         }
     }
+
+    #[test]
+    fn reads_threads() -> Result<(), Box<dyn std::error::Error>> {
+        let barrier = Arc::new(Barrier::new(2));
+
+        let t1_barrier = barrier.clone();
+        let t1_handle = thread::Builder::new()
+            .name("thread-name".to_string())
+            .spawn(move || {
+                t1_barrier.wait();
+            })
+            .unwrap();
+
+        let proc = LinuxTarget::me();
+        let threads = proc.threads()?;
+
+        let threads: Vec<(String, usize)> = threads
```

Comment:
I need to make sure the thread exists before calling this.

Reply:
Maybe use another barrier?

Reply:
fixed